### DISCUSSION TITLE
Devices in Home Assistant randomly in state "unavailable" => mqtt connection issue

### DISCUSSION THREAD

**Discussion by dakky** [arendst/Tasmota#19906] [Q&A] [answered]
Hello

I have the following issue: devices in home assistant are permanently shown as not available. I am using the official addon for tasmota. I tried to debug the issue and it seems, that home assistant sends some status commands:
```
Nov 01 23:05:38 cmnd/tasmota_shelly2/STATUS 1
Nov 01 23:05:38 cmnd/tasmota_shelly2/STATUS 11
Nov 01 23:05:38 cmnd/tasmota_shelly2/STATUS 10
```
followed after about 20secs by a:
```
Nov 01 23:05:59 tele/tasmota_shelly2/LWT Offline
```
So **seems** to happen here? HA asks for same status reports, receives no response and assumes the device is offline (LWT Offline). After some time the tasmota device reports itself back online, HA assumes "device is back" and the cycle repeats:
```
Nov 01 23:06:38 tele/tasmota_shelly2/LWT Online
Nov 01 23:06:38 cmnd/tasmota_shelly2/POWER (null)
Nov 01 23:06:38 tele/tasmota_shelly2/STATE {**redacted sensible content**}
Nov 01 23:06:38 tele/tasmota_shelly2/SENSOR {"Time":"2023-11-01T23:06:17","ANALOG":{"Temperature":47.1},"ENERGY":{"TotalStartTime":"2022-02-13T12:39:15","Total":0.593,"Yesterday":0.005,"Today":0.003,"Period":[0,0],"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":0,"Current":[0.000,0.000]},"Shutter1":{"Position":51,"Direction":0,"Target":51,"Tilt":0},"TempUnit":"C"}
Nov 01 23:06:39 cmnd/tasmota_shelly2/STATUS 1
Nov 01 23:06:39 cmnd/tasmota_shelly2/STATUS 11
Nov 01 23:06:39 cmnd/tasmota_shelly2/STATUS 10
Nov 01 23:07:00 tele/tasmota_shelly2/LWT Offline
``` 

So i tried manually to get a response from tasmota via mqtt:
```
mosquitto_pub -h hostname -t "cmnd/tasmota_shelly2/STATUS" -m "1"
```
I can see this message in mqtt, but NOT in die tasmota console! And still no response. So it looks like tasmota is able to **send** messages to the broker but is not able (does not try?) to receive commands by any other client. 

The network itself seems stable (relativly high signal strength, no package loss when ping from broker => tasmota)

Anyone has an idea what might be wrong here? I do not know where to dig deeper for a solution. 

Oh btw this affects multiple devices, so a hardware fault seems very unlikly. And this setup worked before, the only change i did was moving the mqtt broker to another host and changed the mqtt config on all tasmota devices

**Comment by dakky** [comment]
Update:

I switched on broker on the old host and reconfigured tasmota to use the old mqtt broker. There the mqtt part works like a charm:

```
Nov 01 23:21:16 cmnd/tasmota_shelly2/STATUS 1
Nov 01 23:21:16 stat/tasmota_shelly2/STATUS1 {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:01:24","StartupUTC":"2023-11-01T22:19:51","Sleep":50,"CfgHolder":4617,"BootCount":78,"BCResetTime":"2022-02-13T12:39:15","SaveCount":912,"SaveAddress":"F8000"}}
```

is there something on tasmotas side, which is caching the old mqtt setting? or some other wierd setting I forgot to change besides the hostname of the mqtt broker?

**Comment by Jason2866** [comment]
Tasmota does not "know" anything about the broker, nor does it store anything. It just connects to receive and publish messages.
So your new broker has an issue.

**Reply by barbudor** [reply]
When the old broker is off and Tasmota configured for the new one, does Tasmota console reports connection to the broker?

If yes, and Tasmota still doesn't seems to receive commands through the new broker, are you sure the `FullTopic` is as expected?

You can set `WebLog 3` and restart Tasmota. In the console logs, right after the connection to MQTT and before the INFO messages you should see to which topics Tasmota subscribes to.

**Comment by stefanbode** [comment]
There is a rule you can implement that might mitigate the problem a bit:
`rule2 on Mqtt#Connected do RuleTimer1 0 endon on Mqtt#Disconnected do RuleTimer1 60 endon ON rules#timer=1 DO restart 1 endon`
`rule2 1`

As you see as long as the connect does not work the device start a reboot. with "`wificonfig 0`" you can also enforce a reboot if there is no WLAN connect at the beginning. I do have similar problem from time to time also with not knowing directly what the problem is. I can confirm that during these times also the webbowser on the tasmota is not reachable. so it looks like a wifi problem and not a mqtt problem.

**Reply by dakky** [reply, marked as answer]
I'll resolve this problem: magic.

Over night something fixed this. I now have a stable connection from alle participant. Really no idea what caused this issue and what resolved it. Thanks for trying to help

**Comment by echo-bravo-yahoo** [comment]
I had these exact symptoms, and it turned out to be that my tasmota devices had SetOption104 set to 1 ("never send retained messages"). The discovery mechanism requires retained messages to work after restarting parts of the system. Swapping SetOption104 to 0 on all my tasmota devices resolved it for me.
